File: x-ray-services/src/test/java/com/abien/xray/business/statistics/boundary/MinutelyStatisticsCalculatorTest.scala
package com.abien.xray.business.statistics.boundary

import com.abien.xray.business.store.boundary.Hits
import org.junit._
import Assert._
import org.mockito.Mockito._
import org.scalatest.mock.MockitoSugar
import org.scalatest.junit.JUnitSuite
import org.scalatest.junit.ShouldMatchersForJUnit
import javax.enterprise.event.Event

class MinutelyStatisticsCalculatorTest extends JUnitSuite with MockitoSugar with ShouldMatchersForJUnit {

  var cut: MinutelyStatisticsCalculator = _

  @Before
  def setUp: Unit = {
    cut = new MinutelyStatisticsCalculator()
    cut.hits = mock[Hits]
    cut.minutelyEvent = mock[Event[java.lang.Long]]
  }


  @Test
  def computeRatePerMinute = {
    when(cut.hits.totalHits).thenReturn(10)
    cut.computeStatistics()
    cut.getHitsPerMinute should be("10")
    when(cut.hits.totalHits).thenReturn(12)
    cut.computeStatistics()
    cut.getHitsPerMinute should be("2")
  }

  @Test
  def computeMaxRatePerMinute = {
    when(cut.hits.totalHits).thenReturn(10)
    cut.computeStatistics()
    cut.getMaxHitsPerMinute should be("10")
    when(cut.hits.totalHits).thenReturn(22)
    cut.computeStatistics()
    cut.getMaxHitsPerMinute should be("12")

    when(cut.hits.totalHits).thenReturn(23)
    cut.computeStatistics()
    cut.getMaxHitsPerMinute should be("12")
  }
}

File: x-ray-services/src/test/java/com/abien/xray/business/statistics/boundary/MinutelyStatisticsCalculatorTest.java
/*
 */
package com.abien.xray.business.statistics.boundary;

import java.time.Duration;
import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;
import org.junit.Test;

/**
 *
 * @author airhacks.com
 */
public class MinutelyStatisticsCalculatorTest {

    @Test
    public void computeStatisticsSchedule() {
        Duration duration = Duration.parse("PT" + MinutelyStatisticsCalculator.DURATION);
        assertThat(duration.getSeconds(), is(60l));
    }

}
